### PR TITLE
Handle security issue with OT extension

### PR DESCRIPTION
**Pull request by jot2re**
This closes issues #394 

The PR also includes minor changes suggested by the SonarLint plugin in order to improve the general code quality in the files touched.

### CODE REVIEW

File: tools/mascot/src/main/java/dk/alexandra/fresco/tools/mascot/MascotResourcePoolImpl.java [hunk 1]
```diff
@@ -90,16 +87,15 @@ public FieldElementPrg getLocalSampler() {
...
-      throw new IllegalArgumentException("Cannot initialize with self");
-    }
-    CoinTossing ct = new CoinTossing(getMyId(), otherId, getRandomGenerator());
-    ct.initialize(network);
-    OtExtensionResourcePool otResources = new OtExtensionResourcePoolImpl(getMyId(), otherId,
-        getPrgSeedLength(), getLambdaSecurityParam(), getInstanceId(),
-        getRandomGenerator(), ct, seedOts.get(otherId));
-    return new BristolRotBatch(new RotFactory(otResources, network),
-        getPrgSeedLength(), getLambdaSecurityParam());
+      if (getMyId() == otherId) {
```
Comment (jonas-lj) [inline tools/mascot/src/main/java/dk/alexandra/fresco/tools/mascot/MascotResourcePoolImpl.java:90]: Use the correct style

File: tools/ot/src/main/java/dk/alexandra/fresco/tools/ot/otextension/RotFactory.java [hunk 2]
```diff
@@ -18,17 +18,21 @@ public class RotFactory {
    * @param network The network instance
    */
   public RotFactory(OtExtensionResourcePool resources, Network network) {
-    this.resources = resources;
-    this.network = network;
+      this.resources = resources;
```
Comment (jonas-lj) [inline tools/ot/src/main/java/dk/alexandra/fresco/tools/ot/otextension/RotFactory.java:21]: Style

File: tools/ot/src/main/java/dk/alexandra/fresco/tools/ot/otextension/RotFactory.java [hunk 3]
```diff
@@ -18,17 +18,21 @@ public class RotFactory {
...
-  }
+    public OtExtensionResourcePool getResources() {
+        return resources;
+    }
 
-  public RotReceiver createReceiver() {
-    CoteReceiver receiver = new CoteReceiver(resources, network);
-    return new RotReceiverImpl(receiver, resources, network);
-  }
+    public RotSender createSender() {
```
Comment (jonas-lj) [inline tools/ot/src/main/java/dk/alexandra/fresco/tools/ot/otextension/RotFactory.java:29]: Style

File: tools/ot/src/test/java/dk/alexandra/fresco/tools/ot/otextension/BristolRotBatchDemo.java [hunk 4]
```diff
@@ -51,10 +52,10 @@ public void runPartyTwo(int pid) throws IOException {
         lambdaSecurityParam);
     OtExtensionResourcePool resources = ctx.createResources(1);
     RotBatch rotBatch = new BristolRotBatch(new RotFactory(ctx.createResources(1), ctx
-            .getNetwork()), resources.getComputationalSecurityParameter(),
-        resources.getLambdaSecurityParam());
+            .getNetwork())
+    );
     List<Pair<StrictBitVector, StrictBitVector>> messages = rotBatch.send(
-        amountOfOTs, messageSize);
+            amountOfOTs, messageSize);
```
Comment (jonas-lj) [inline tools/ot/src/test/java/dk/alexandra/fresco/tools/ot/otextension/BristolRotBatchDemo.java:58]: Style

File: tools/ot/src/test/java/dk/alexandra/fresco/tools/ot/otextension/TestCote.java [hunk 5]
```diff
@@ -1,18 +1,18 @@
...
 import dk.alexandra.fresco.framework.util.Drbg;
 import dk.alexandra.fresco.framework.util.StrictBitVector;
 import dk.alexandra.fresco.tools.cointossing.CoinTossing;
 import dk.alexandra.fresco.tools.helper.HelperForTests;
-
-import java.lang.reflect.Field;
 import org.junit.Before;
 import org.junit.Test;
 
+import java.lang.reflect.Field;
```
Comment (jonas-lj) [inline tools/ot/src/test/java/dk/alexandra/fresco/tools/ot/otextension/TestCote.java:12]: Style

File: tools/ot/src/test/java/dk/alexandra/fresco/tools/ot/otextension/TestFunctionalBristolOt.java [hunk 6]
```diff
@@ -125,44 +126,44 @@ public void testBristolOt() {
 
   private List<Pair<StrictBitVector, StrictBitVector>> bristolRotBatchSend(
       OtExtensionTestContext ctx, int batchSize, int id) {
-    OtExtensionResourcePool resources = ctx.createResources(id);
-    BristolRotBatch rotBatchSender =
-        new BristolRotBatch(new RotFactory(resources, ctx.getNetwork()),
-            resources.getComputationalSecurityParameter(), resources.getLambdaSecurityParam());
-    List<Pair<StrictBitVector, StrictBitVector>> messages =
-        rotBatchSender.send(batchSize, messageLength);
+      OtExtensionResourcePool resources = ctx.createResources(id);
```
Comment (jonas-lj) [inline tools/ot/src/test/java/dk/alexandra/fresco/tools/ot/otextension/TestFunctionalBristolOt.java:129]: Style

**Review by jonas-lj — CHANGES_REQUESTED** [review]
Make sure to use the correct code style for all changed files. Otherwise it looks good.

**Comment by jot2re** [comment]
@jonas-lj requested changes have been added.

**Review by jonas-lj — APPROVED** [review]

**Pull request merged**